brave/brave-core
# Deprecate local node support and ipfs scheme

Resolves https://github.com/brave/brave-browser/issues/37735



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

**iOS IPFS Deprecation Test Plan:**
1. open Brave iOS
2. Tap `...`
3. Go to Settings/Web3
4. Observe that no IPFS related settings are displayed in the screen
5. Make sure all Web3 Domain settings are all set to `Ask`
6. Go back to home page
7. input an IPFS scheme url in omnibox. For example: ipfs://QmX4nfgA35MiW5APoc4P815hMcH8hAt7edi5H3wXkFm485/1688.html
8. No IPFS configuration interstitial page will be loaded. and IPFS url will not be resolved and loaded. A search will be performed instead.
9. Input a SNS url in omnibox. For example: http://onybose.sol/
10. An interstitial page will be loaded. Tap `Proceed using an SNS server`
11. Observe the SNS url is resolved and loaded
12. Go back to Settings/Web3. Observe `Resolve SNS Domain Names` is now set to `Enabled`
13. Change that to `Ask` abd go back to home page
14. Repeat step 9 and 10 but tap `Disable`
15. Observe an error page is loaded
17. Go back to Settings/Web3. Observe `Resolve SNS Domain Names` is now set to `Disabled`
18. Go back to home page and input a ENS url in omnibox. For example: http://offchainexample.eth/ (the resolved url is an IPFS scheme url)
19. An interstitial page will be loaded. Tap `Proceed using infura server`
20. Another interstitial page will be loaded. Tap `Proceed with offchain lookup`
21. observe the ENS is now resolved and loaded. The process is ENS -> offchain lookup -> IPFS url resolved. (Even though IPFS is deprecated. but for IPFS url that was the result of other Web3 Domain Names. BraveCore will resolve those IPFS urls using the public gateway https://ipfs.io/)
22. Go to Settings/Web3. Observe `Resolve ENS Domain Names` and `Allow ENS Offchain Lookup` are now set to `Enable`
23. Change them to `Ask` and go back home page 
24. Repeat step 18 and 19 but tap `Disable` 
25. Observe an error page is loaded
26. Change `Resolve ENS Domain Names` to `Enabled` in Settings/Web3
27. Go back to home and reload the error page
28. Another interstitial page will be loaded. Tap `Disable`
29. Observe an error page is loaded
30. Input an unstoppable domains in omnibox. For example: jim-unstoppable.crypto
31. Another interstitial page will be loaded. Tap `Proceed using infura server`
32. Observe the url is now resolved using the public gateway https://ipfs.io/
33. Go back to Settings/Web3 and disable `Resolve Unstoppable Domains Domain Names`
34. Repeat step 30
35. Observe an error page is loaded
